File: iosgeneratedcode/jcrystal/mobile/entities/enums/Categories.swift
public enum Categories : Int{
	case MEN = 1000
	case WOMEN = 2000
	case KIDS = 3000
	var name: String?{
		switch self{
			case .MEN : return "Men"
			case .WOMEN : return "Women"
			case .KIDS : return "Kids"
		}
	}
	var rawName: String?{
		switch self{
			case .MEN : return "MEN"
			case .WOMEN : return "WOMEN"
			case .KIDS : return "KIDS"
		}
	}
	var id: Int{
		return self.rawValue
	}
	public static func fromId(_ id : Int) -> Categories{
		return Categories(rawValue: id)!
	}
	static let values = [MEN, WOMEN, KIDS]
}

File: src/main/java/dev/jcrystal/crystalfash/entities/Categories.java
package dev.jcrystal.crystalfash.entities;


public enum Categories {
	MEN(1000, "Men"),
	WOMEN(2000, "Women"),
	KIDS(3000, "Kids");
	public final long id;
	public final String name;
	private Categories(long id, String name) {
		this.id = id;
		this.name = name;
	}
	public static Categories fromId(int id) {
		for (Categories val: Categories.values()) {
			if (val.id==id) {
				return val;
			}
		}
		return null;
	}
}